### ISSUE TITLE
C language has incorrect comments

### ISSUE THREAD

**Issue by NN---** [notepad-plus-plus/notepad-plus-plus#484]
``` c
a
```

produces

``` c
// a
```

Which is not correct in ANSI-C.
It must be with /\* */

I propose adding C99 which will use the one line comment.

**Comment by Codextor** [comment]
I know this issue is old, but just out of curiosity, could someone provide a reference link which states that single line comments in C should use /* */ notation and not //

From what I could find online, the [C11](http://www.open-std.org/JTC1/SC22/WG14/www/docs/n1570.pdf) & [C17](http://www.open-std.org/jtc1/sc22/wg14/www/abq/c17_updated_proposed_fdis.pdf) standards accepts both ways of commenting.
_Refer to section 6.4.9 on comments_

**Comment by sasumner** [comment]
@Codextor I believe that a long time ago, C did not accept // comments.  Thus there was no option, you had to use the other notation everywhere, single line or not.  I don't think you're going to find any kind of reference like you are looking for.

**Comment by Codextor** [comment]
OK so prior to C99, we didn't have // comments in the C language. Hence if someone uses the default notepad++ method of commenting single lines, and tries to compile the code with an older compiler, he might end up with errors.
Any plans of fixing this issue?

**Comment by sasumner** [comment]
@Codextor I can't imagine so...  1999-ish was a long time ago.

**Comment by bpersonick** [comment]
@Sandeep

TL;DR:

  C comments do not allow // prior to C99  when the C++ feature was
back-ported into the C language ( See:
https://developerinsider.co/c-programming-language-version-history/,
https://en.wikipedia.org/wiki/C_(programming_language)#Syntax, and
https://en.wikipedia.org/wiki/C_syntax#Comments )

 The True C Programmer is the one most likely to be opening and editing C
Code in Notepad ++, while C++ and C# code is more likely to be developed
elsewhere and use Notepad ++  to make quick review of or edits to that code
(Although that is not to say it is uncommon for some C++ or C# Developer to
utilize NP++ for all of their work!).

 It can be shown that there is a real and perceived value to a separation
of the C Syntax and Highlighting from that of C++ and C# within the
built-in syntax highlighting and commenting of NP++.

 This would benefit those who truly write C code as they often maintain
legacy systems requiring the code be compiled under truly old versions of
the C compiler to ensure functionality on the target systems, and could
easily be implemented as a global preference, and a per file toggle by
separating the Syntaxes within NP++ and adding such preference guidelines
on which to use when opening .c and .h files (vs file types only associated
with the newer languages such as .cpp which would not need such a
preference)

 More info and thoughts follow:

================

maiantain the and HoweverTrue C Comments should follow the /* */ method for
compatibility as C++ is not a true superset of C, and should be treated as
a separate syntax instead of being lumped in with C++ and C#.


  The single Line comment using // was introduced as a feature in the first
version of C++ in 1983,  It was not was backported as a feature into C
until 1999 as part of the C99 Standard.

  So from the perspective of compatibility, one-line comments using
/*[LINE(S)]*/ may appear in any code for any version of C, C++, or C# as
the /* */ feature allows any length comment.

   // also, unline /* */ always flows to the end of the line
where implemented, while /* */ can be used to start or end mid-line
commenting only a sub-portion of code on a given line.

  However /* */ does not nest, so commenting a section with comments inside
of it will not work, leading to needing to create several sets of comments,
or situ*ations where programmers will generate IF s*tructures around the
area to be commented (eg IF 1==0 { THis is all /* commented*/ hahahahaha!}
defeating the purpose of the comment feature in the first place.

 This is the need that the // style comment was designed to address, as the
only ending operator is the carriage return, you may use it to comment out
large blocks of code, which may themselves have /* */ style or // style
comments inside of them without causing a conflict in the commenting
requiring you to spend a lot of time fix umteen hours debugging the comment
itself or cludging in your own IF $false = $true style comments.

  It is only on rare occasions you might be looking at old code, and the
syntax highlighting will respect either style comments, so that is not an
issue.

  The only thing which may be an issue is edge cases where programmers need
to use old compilers prior to C99 to compile their code.  If the programmer
may compile the code in at least the C99 compiler with confidence that the
resulting executable will run on the system they plan to use it on, then
they have no worries.

  So the only time a requirement to use /* */ arises reasonably is when any
developer to this day wishes to make a within-line comment, a multi-line
comment that ends mid-line (as any comment may start mid-line, and the edge
case scenario where someone is maintaining systems which are running
decades-old code (Which is an admirable task, and one that is far more
common than many would believe, but is not the normal position of a
developer.)

  That stated, it is far more likely for today's Developer to be writing in
C# or C++ and using a modern compiler there-of or using a C++ or C#
compiler.ng Code in C, as opposed to C++ or C# is expressly doing so to
maintain compatible code for older systems, and would likely benefit from
having the Syntax highlighting and commenting separated from the C++and C#
methods instead of having this agglomerated styling and commenting class.

  This is especially true as today, those who are truly programming in C
(as opposed to C++ or C#) are inherently in the practice of maintaining
legacy systems which likely require them to utilize specific compilers to
write and amend code which will be compatible on these systems.

  As C++ and C# have many viable IDEs and syntax highlighting methods,
directly and indirectly, available to them, it is actually the C programmer
who would have the most incentive to utilize Notepad++ to review and write
code, while the C++ or C# programmer is more incentivized to use notepad++
for quick edits and review of code when they don't wnst need a full burden
of other tools to be in the way.


  These facts support the Idea of requiring a syntax and commenting
difference be available as a default option within the in-built Syntax and
commenting methods for C/C++/C# as their agglomeration into a single bucket
has hindered the true C programmer, the individual most in need of a
product such as Notepad ++ to modernize their work,

  It is therefore appropriate, that we consider a method of separating C
syntax Highlighting and commenting from that of C++ or C#, in a way that
may be easily controllable as a global level preference, and toggle-able at
a per file instance

**Comment by Codextor** [comment]
Thanks @bpersonick for the detailed explanation. I understand the issue clearly now.
